CRM/Grant/Form/Grant.php

```diff
@@ -85,6 +85,13 @@ public function preProcess() {
       CRM_Core_Error::fatal(ts('You do not have permission to access this page.'));
     }
 
+    $this->userDisplayName = $this->userEmail = NULL;
+    if ($this->_contactID) {
+      list($this->userDisplayName,
+        $this->userEmail
+        ) = CRM_Contact_BAO_Contact_Location::getEmailDetails($this->_contactID);
+      $this->assign('displayName', $this->userDisplayName);
```

Comment:
I think there is no need to have class variables userDisplayName and userEmail as these are just fetched to assign displayName value to tpl. So can we have it like:
```php
if ($this->_contactID) {
 list($displayName, $email)  = CRM_Contact_BAO_Contact_Location::getEmailDetails($this->_contactID);
 $this->assign('displayName', $displayName);
}
```

Reply:
It would be nice to have a helper IMHO - 

CRM_Core_Form could have assignDisplayNameFormContactEmail($contactID) or similar